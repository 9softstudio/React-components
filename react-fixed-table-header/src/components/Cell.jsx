import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Cell extends Component {
    constructor(props) {
        super(props);
    }

    static propTypes = {
        header: PropTypes.bool
    }

    static defaultProps = {
        header: false
    }

    render() {
        const { header, colWidth, ...rest } = this.props;
        return (
            header ? <th {...rest}>{this.props.children}</th> : <td {...rest}>{this.props.children}</td>
        );
    }
}

export default Cell;