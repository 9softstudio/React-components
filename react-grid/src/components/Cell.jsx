import React from 'react';

const Cell = (props) => {
    return (<td {...props}>{props.children}</td>)
}


export default Cell;