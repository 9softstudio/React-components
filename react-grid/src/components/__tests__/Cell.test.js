import React from 'react'
import Cell from '../Cell'
import renderer from 'react-test-renderer'

test('Cell component render correctly', () => {
    const cell = renderer.create(
        <Cell>Cell Content</Cell>
    )

    const tree = cell.toJSON();

    expect(tree).toMatchSnapshot();
})

test('Cell component render cell correctly with html attributes', () => {
    const cell = renderer.create(
        <Cell className="cell" style={{ width: 100, height: 200 }}>Cell Content</Cell>
    )

    const tree = cell.toJSON();

    expect(tree).toMatchSnapshot();
})

test('Cell component render header cell correctly with html attributes', () => {
    const cell = renderer.create(
        <Cell className="cell" style={{ width: 100, height: 200 }} header={true}>Cell Content</Cell>
    )

    const tree = cell.toJSON();

    expect(tree).toMatchSnapshot();
})

test('Cell component render sortable header cell correctly with html attributes', () => {
    const cell = renderer.create(
        <Cell className="cell" style={{ width: 100, height: 200 }} header={true} sortable={true} sorting={true} asc={true}>Cell Content</Cell>
    )

    const tree = cell.toJSON();

    expect(tree).toMatchSnapshot();
})