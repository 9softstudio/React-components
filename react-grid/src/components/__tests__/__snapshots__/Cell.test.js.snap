// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell component render cell correctly with html attributes 1`] = `
<td
  className="cell"
  style={
    Object {
      "height": 200,
      "width": 100,
    }
  }
>
  Cell Content
</td>
`;

exports[`Cell component render correctly 1`] = `
<td>
  Cell Content
</td>
`;

exports[`Cell component render header cell correctly with html attributes 1`] = `
<td
  className="cell"
  header={true}
  style={
    Object {
      "height": 200,
      "width": 100,
    }
  }
>
  Cell Content
</td>
`;

exports[`Cell component render sortable header cell correctly with html attributes 1`] = `
<td
  asc={true}
  className="cell"
  header={true}
  sortable={true}
  sorting={true}
  style={
    Object {
      "height": 200,
      "width": 100,
    }
  }
>
  Cell Content
</td>
`;
